frontend/src/components/NewRunParametersV2.tsx

```diff
@@ -134,10 +156,17 @@ function NewRunParametersV2(props: NewRunParametersProps) {
                 onChange={value => {
                   const nextUpdatedParameters: RuntimeParameters = {};
                   Object.assign(nextUpdatedParameters, updatedParameters);
-                  nextUpdatedParameters[k] = { string_value: value };
+                  nextUpdatedParameters[k] = value;
                   setUpdatedParameters(nextUpdatedParameters);
                   if (props.handleParameterChange) {
-                    props.handleParameterChange(nextUpdatedParameters);
+                    let parametersInRealType: RuntimeParameters = {};
+                    Object.entries(nextUpdatedParameters).map(([k, v1]) => {
+                      parametersInRealType[k] = convertInput(
+                        v1.toString(),
```

Comment:
Added `toString()` here to avoid error caused by non-string type default value, because the `convertInput` function only accept `string` type input.

Reply:
`updatedParameters` is supposed to hold the real type value. Can we simplify the logic here to the folloiwng?

```
onChange={value => {
        const nextUpdatedParameters: RuntimeParameters = {};
        Object.assign(nextUpdatedParameters, updatedParameters);
        nextUpdatedParameters[k] = convertInput(value, v.parameterType);
        setUpdatedParameters(nextUpdatedParameters);
        if (props.handleParameterChange) {
                  props.handleParameterChange(nextUpdatedParameters);
        }
}}

Reply:
I tried this before; however, it has bugs (the TextField won't be updated, or I should say TextField will be updated to `null` if user type invalid input).
For example, when users try to type `true`, the only way that TextField can be updated is copy `true` and directly paste into the box, because `t`, `tr`, `tru` are all invalid inputs for boolean type, which will get `null` from `convertInput`. This bugs will make the users **not** see what they have typed.

https://user-images.githubusercontent.com/56132941/176970710-b40fef0d-fe40-4eb9-9c01-262ef77d9bf3.mov

Reply:
I see, then it makes sense to convert input only for  `handleParameterChange`'s use.